Add Azure integration (#65)
port-labs/Port-Ocean: integrations/azure/README.md

> # Azure

Reviewer: More documentation please about how do i know which resources are supported and how to map

> # Azure

Reviewer: Doc the installation process

> # Azure

Reviewer: Talk to sinai
If im not wrong it can be deployed using azure
So it shouldnt show the helm installation

  - **Permissions** - The integration will need to have the following permissions: (An example will be provided later on in the documentation)
    - **Actions**: You'll need to specify the [actions](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-definitions#actions) you want to grant the integration.

#### Azure Infrastructure

- **Azure Subscription**: You'll need an Azure subscription to deploy the integration.
- **Azure Resource Group**: The integration need to be deployed to an Azure Resource Group, you can pass an existing one, or the integration will create a new one. (To override add `resource_group_id: <your-resource-group-id>`)
- **Azure Container App**: The integration will be deployed using Azure Container App. Which requires some extra infrastructure to be deployed, by default if not specified otherwise we will deploy the associate infrastructure.
  - **Azure Log Analytics Workspace**: The integration will create a new Log Analytics Workspace to store the logs of the integration. (To override add `log_analytics_workspace_id: <your-log-analytics-workspace-id>`)

>   - **Azure Container App Environment**: The integration will create a new Container App Environment to deploy the integration to. (To override add `container_app_environment_id: <your-app-environment-id>`)

Reviewer 2: Add where?

#### Azure Event Grid
- **Azure Event Grid System Topic**: To allow the integration to receive events from Azure, an Event Grid System Topic of type `Microsoft.Resources.Subscriptions` will be needed. The integration will create a new System Topic and will subscribe to it. (Due to a limitation in Azure only one Event Grid System Topic of type `Microsoft.Resources.Subscriptions` can be created per subscription, so if you already have one you'll need to pass it to the integration using `event_grid_system_topic_name: <your-event-grid-system-topic-name>`) Further example on how to create the System Topic will be provided later on in the documentation.
- **Azure Event Grid Subscription**: To Pass the events from the System Topic to the integration, an Event Grid Subscription will be needed. The integration will create a new Subscription and will pass the events to the integration.

## Installation

The integration is deployed using Terraform on Azure [ContainerApp](https://learn.microsoft.com/en-us/azure/container-apps/overview).

> The integration is being triggered in two ways:

Reviewer 2: ```suggestion
The integration can be triggered in two ways:
```

- On events sends from the Azure Event Grid.
- When a change in the integration configuration is detected.

> 

Reviewer 2: To deploy the integration, run the following commands:

        apiVersion: '2023-01-01'
    port:
      entity:
        mappings:
          # lower case the resource group namespace and name to align with other azure resources
          identifier: '.id | split("/") | .[3] |= ascii_downcase |.[4] |= ascii_downcase | join("/")'
          title: .name
          blueprint: '"storageContainer"'
          properties:
            publicAccess: .properties.publicAccess
            hasImmutabilityPolicy: .properties.hasImmutabilityPolicy
            hasLegalHold: .properties.hasLegalHold
            deleted: .properties.deleted
            deletedTime: .properties.deletedTime
            remainingRetentionDays: .properties.remainingRetentionDays
            leaseStatus: .properties.leaseStatus
            leaseState: .properties.leaseState
            defaultEncryptionScope: .properties.defaultEncryptionScope
            version: .properties.version
          relations:
            # lower case the resource group namespace and name to align with other azure resources
            storageAccount: '.id | split("/") | .[3] |= ascii_downcase | .[4] |= ascii_downcase | .[:-4] |join("/")'
```

</details>

## Adding a new Azure Resource kind

> 

Reviewer 2: I suggest adding a summary of the steps that need to be taken (blueprint, mapping, auth, etc)

### Authorization 
When adding new Azure resource we need to make sure our exporter has permissions to query it.
To do so we will need to add a new action permission to the integration role definition.
- The action permission should be added to the `actions` section of the integration role definition.

### Event Grid
When adding new Azure resource we need to make sure our exporter will be able to receive events related to it.
To do so we will need to add a new event filter to the integration event grid subscription. For more information on [action permissions](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-definitions#actions)

### Example
Here is an example on how to apply it, as in the base example we will head to the `deployment/azure/container_app/examples` directory
and apply the terraform changes.

`Microsoft.Network/virtualNetworks` is the resource we want to add.
```sh
export TF_VAR_port_client_id=<PORT_CLIENT_ID>
export TF_VAR_port_client_secret=<PORT_CLIENT_SECRET>

git clone git@github.com:port-labs/Port-Ocean.git

cd deployment/azure/container_app/examples

terraform init

> terraform apply -var='subscription_id=<subscription_id>' -var='action_permissions_list=["Microsoft.app/containerapps/read","Microsoft.Storage/storageAccounts/read","Microsoft.ContainerService/managedClusters/read","Microsoft.Network/loadBalancers/read","Microsoft.Resources/subscriptions/resourceGroups/read","Microsoft.Resources/subscriptions/resources/read", "Microsoft.Network/virtualNetworks/read"]' -var='event_grid_event_filter_list=["Microsoft.App/containerApp","Microsoft.Storage/storageAccounts","Microsoft.Compute/virtualMachines","Microsoft.Network/loadBalancers","Microsoft.Resources/subscriptions/resourceGroups", "Microsoft.Network/virtualNetworks]'

Reviewer 2: No one will ever read this super long line
